addons/road-generator/nodes/road_lane.gd

```diff
@@ -151,10 +151,12 @@ func unregister_vehicle(vehicle: Node) -> void:
 
 ## Return all vehicles registered to this lane, performing cleanup as needed.
 func get_vehicles() -> Array:
+	var valid_vehicles: Array = []
 	for vehicle in _vehicles_in_lane:
-		if (not is_instance_valid(vehicle)) or vehicle.is_queued_for_deletion():
-			_vehicles_in_lane.erase(vehicle)
-			continue
+		if is_instance_valid(vehicle) and not vehicle.is_queued_for_deletion():
+			valid_vehicles.append(vehicle)
+	
+	_vehicles_in_lane = valid_vehicles
```

Comment:
Curious about this change - was this causing problems for you?

The original idea is that we perform lazy garbage detection by only cleaning up references once trying access them, hence being deferred to this point. Without this, references risk accumulating over time if the vehicle actor doesn't clean itself up (they should, but this is a nice safety fallback).